libr/bin/format/elf/elf.c

```diff
@@ -3546,21 +3569,18 @@ static RBinElfSymbol* Elf_(_r_bin_elf_get_symbols_imports)(ELFOBJ *bin, int type
 						ret[ret_ctr].name[0] = 0;
 					} else {
 						bool found = false;
-						j = -1;
-						while (!ret[++j].last && j < prev_ret_size) {
-							if (ret[j].offset == ret[ret_ctr].offset &&
-									strcmp (ret[j].name, "") != 0 && strcmp (ret[j].name, &strtab[st_name]) == 0
-									&& strcmp (ret[j].type, type2str (NULL, NULL, &sym[k])) == 0) {
-								found = true;
-								break;
-							}
+						RBinElfSymbol *previous = ht_pp_find (symbols_by_name, &strtab[st_name], NULL);
```

Comment:
Is a HT by name enough? I *think* you could have multiple symbols with the same name but different properties (e.g. offset, type, etc.) and you would have to consider them differently.

In theory, we could still use a HtPP with a custom HashFunction that would accept as key name+offset+type.

Reply:
This is the same discussion as one in comment bellow, right?